http/pattern.go

```diff
@@ -0,0 +1,206 @@
+// Copyright 2023 The Go Authors. All rights reserved.
```

Comment:
Please add this comment at the top of the file:

// TINYGO: The following is copied and modified from Go 1.23.7 official implementation.